interfaces/builtin/avahi_control.go

```diff
@@ -135,33 +138,27 @@ func (iface *avahiControlInterface) AppArmorConnectedPlug(spec *apparmor.Specifi
 }
 
 func (iface *avahiControlInterface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
-	if !release.OnClassic {
-		// NOTE: this is using avahi-observe permanent slot as it contains
-		// base declarations for running as the avahi service.
-		spec.AddSnippet(avahiObservePermanentSlotAppArmor)
-	}
+	// NOTE: this is using avahi-observe permanent slot as it contains
+	// base declarations for running as the avahi service.
+	spec.AddSnippet(avahiObservePermanentSlotAppArmor)
```

Comment:
The PermanentSlot policy should not be applied when the slotting snap is core, core18, etc.